biodatasets/epicqa/epicqa.py

```diff
@@ -0,0 +1,336 @@
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        my_urls = _URLs[self.config.name]
+        data_dir = dl_manager.download_and_extract(my_urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_dir,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """Yields examples as (key, example) tuples."""
+
+        if self.config.subset_id.endswith("research"):
+            subset_name = "research"
+        elif self.config.subset_id.endswith("consumer"):
+            subset_name = "consumer"
+        else:
+            raise ValueError(f"subset_id {self.config.subset_id} does not end with 'research' or 'consumer'")
+
+        uid = 0
+
+        # For each sub directory
+        for sub_dir in _DIRS[subset_name]:
+
+            # Get the JSON file
+            for f_path in os.listdir(filepath + "/" + sub_dir):
+
+                # Check if JSON file
+                if ".json" not in f_path:
+                    continue
+
+                # Get the JSON file path
+                full_path = filepath + "/" + sub_dir + f_path
+
+                # Read the JSON file content
+                with open(full_path, encoding="utf-8") as file:
+
+                    # Parse it and add it to the data array
+                    doc = json.load(file)
+
+                    # Generate source schema for the consumers articles
+                    if self.config.name == "epicqa_consumer_source":
+
+                        yield uid, {
+                            "document_id": doc["document_id"],
+                            "metadata": {
+                                "title": doc["metadata"]["title"],
+                                "url": doc["metadata"]["url"],
+                            },
+                            "contexts": [
+                                {
+                                    "section": c["section"],
+                                    "context_id": c["context_id"],
+                                    "text": c["text"],
+                                    "sentences": [
+                                        {
+                                            "sentence_id": s["sentence_id"],
+                                            "start": s["start"],
+                                            "end": s["end"],
+                                        }
+                                        for s in c["sentences"]
+                                    ],
+                                }
+                                for c in doc["contexts"]
+                            ],
```

Comment:
could this just be `yield uid, doc` ?

Reply:
I have done it before, but I got issues with the JSON dictionary, like I said on discord.